### PR TITLE
Update docker.rb

### PR DESCRIPTION
**Pull request by miccal**
See https://github.com/caskroom/homebrew-cask/pull/30586 for reference.

### CODE REVIEW

**Review by vitorgalvao — CHANGES_REQUESTED** [review]
If the `app` has retained its name, the cask token must remain what it was. The token reference is imperative.

**Comment by miccal** [comment]
Sorry @vitorgalvao I always confuse the token reference with the `name` not the `app`.

File: Casks/docker.rb [hunk 1]
```diff
@@ -1,11 +1,10 @@
 cask 'docker' do
-  version '1.13.1.15353'
-  sha256 '137ac8f191f043759f09fef705219d97586cac1e742f34715d823aefca2a2591'
+  version :latest
+  sha256 :no_check
 
-  url "https://download.docker.com/mac/stable/#{version}/Docker.dmg"
-  appcast 'https://download.docker.com/mac/stable/appcast.xml',
```
Comment (vitorgalvao) [inline Casks/docker.rb]: The `appcast` seems to still be functional. It mentions `17.03.0-ce-mac1 (15583)`, so it seems correct.

Comment (miccal) [inline Casks/docker.rb]: I swear it did not say that when I checked it a few hours ago!

Comment (miccal) [inline Casks/docker.rb]: I also did not see the Github release links before.

File: Casks/docker.rb [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
-  version '1.13.1.15353'
-  sha256 '137ac8f191f043759f09fef705219d97586cac1e742f34715d823aefca2a2591'
+  version '17.03.0'
+  sha256 '17a05db62267b175986536594b1a4ea0a393690031d0ae688f6db73e2399d30a'
 
-  url "https://download.docker.com/mac/stable/#{version}/Docker.dmg"
-  appcast 'https://download.docker.com/mac/stable/appcast.xml',
-          checkpoint: '166a08c5e8877e645850e78720c735a9492b6f75e0d9e81da1331c6521eba8a5'
-  name 'Docker'
+  url "https://get.docker.com/builds/Darwin/x86_64/docker-#{version}-ce.tgz"
```
Comment (vitorgalvao) [inline Casks/docker.rb]: Gotten from the github releases page you found.

File: Casks/docker.rb [hunk 3]
```diff
@@ -1,11 +1,12 @@
 cask 'docker' do
-  version '1.13.1.15353'
-  sha256 '137ac8f191f043759f09fef705219d97586cac1e742f34715d823aefca2a2591'
+  version '17.03.0'
+  sha256 '17a05db62267b175986536594b1a4ea0a393690031d0ae688f6db73e2399d30a'
 
   url "https://download.docker.com/mac/stable/#{version}/Docker.dmg"
```
Comment (vitorgalvao) [inline Casks/docker.rb]: Missed that this one was already version. Also, the one I added was not the same thing.

Comment (miccal) [inline Casks/docker.rb]: Yes, but it no longer works.

**Pull request merged**